src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/JsonLoginFactory.php

```diff
@@ -24,6 +24,8 @@
  */
 class JsonLoginFactory extends AbstractFactory implements AuthenticatorFactoryInterface
 {
+    public const PRIORITY = FormLoginFactory::PRIORITY;
```

Comment:
The fact that `JsonLoginFactory` is absolutely not linked to `FormLoginFactory` makes me wonder:
Would it make sense to put these predefined, well-known priority constants at a higher level of abstraction? e.g.
```php
interface AuthenticatorFactoryInterface {
    public const PRIORITY_FORM = -30;
    // ...
}
```
Or do we really want to get rid of these well-known categorized positions in favor of arbitrary integers?

Reply:
I think we should maybe give json a dedicated priority (note that the "positions" covered all authn listeners available in Symfony 2).

I added this constant, and want to propose it for other listeners, to allow developers to say "I want to be executed just before form login" by doing `return FormLoginFactory::PRIORITY + 1;`